server/manager/src/main/java/org/apache/accumulo/manager/upgrade/Upgrader9to10.java

```diff
@@ -144,6 +150,11 @@ public void upgradeMetadata(ServerContext ctx) {
     upgradeFileDeletes(ctx, Ample.DataLevel.USER);
   }
 
+  @Override
+  public void upgradeFiles(ServerContext ctx) {
+    dropSortedMapWALFiles(ctx.getVolumeManager());
```

Comment:
What happens when we delete these files? Wouldn't sorted files have references in the metadata tables? And, if so, would their original WAL files have already been subject to deletion? Is it safe to just remove these and Accumulo will just recreate them using the new format? That behavior isn't clear to me.

Reply:
> What happens when we delete these files? 

Accumulo will automatically re-sort the WALs based off of the original files in `/accumulo/wal`.

> Wouldn't sorted files have references in the metadata tables? 

No they do not. The references are stored in ZK and hdfs. The `DistributedWorkQueue` runs the `LogSorter` based on data in the ZK node. Each node is populated based on the WAL filename/ID taken from `/accumulo/wal`. This method only deletes the temporary sorted files in `/accumulo/recovery`.

> Is it safe to just remove these and Accumulo will just recreate them using the new format? 

Yes that is the intent. Without this change, Accumulo won't be able to recovery tablets in the old map files. If it happens that metadata tablets are in these old sorted map files, then Accumulo won't be able to start.

Reply:
Here is some sample logging that might help:
<pre>
manager_ip-10-113-12-25.log:2021-07-08T09:03:05,079 [recovery.RecoveryManager] INFO : Created zookeeper entry /accumulo/5012a7fa-5daa-45fc-92ca-1c789e70d4f7/recovery/13315719-9dee-460e-92ab-7a3276d34a26 with data hdfs://localhost:8020/accumulo/wal/ip-10-113-12-25+9997/13315719-9dee-460e-92ab-7a3276d34a26|hdfs://localhost:8020/accumulo/recovery/13315719-9dee-460e-92ab-7a3276d34a26
manager_ip-10-113-12-25.log:2021-07-08T09:03:06,693 [recovery.RecoveryManager] INFO : Starting recovery of hdfs://localhost:8020/accumulo/wal/ip-10-113-12-25+9997/13315719-9dee-460e-92ab-7a3276d34a26 (in : 20s), tablet +r<< holds a reference
tserver_ip-10-113-12-25.err:2021-07-08T09:03:05,093 [log.LogSorter] INFO : Copying hdfs://localhost:8020/accumulo/wal/ip-10-113-12-25+9997/13315719-9dee-460e-92ab-7a3276d34a26 to hdfs://localhost:8020/accumulo/recovery/13315719-9dee-460e-92ab-7a3276d34a26
tserver_ip-10-113-12-25.err:2021-07-08T09:03:06,587 [log.LogSorter] INFO : Finished log sort 13315719-9dee-460e-92ab-7a3276d34a26 119262720 bytes 2 parts in 1501ms
</pre>

Reply:
Oh, I thought we put references to them in the "log" column family in the metadata table. Looking at the code, it looks like we use that column family for storing references to unsorted logs instead, when a tablet is unassigned from a dead tserver. I knew sorted WALs were short-lived, but I didn't expect them to be not persisted in the metadata.